src/app_service/service/wallet_account/service.nim

```diff
@@ -603,6 +589,40 @@ QtObject:
       error "error: ", procName="updateWalletAccount", errName=e.name, errDesription=e.msg
     return false
 
+  proc updateWalletAccountProdPreferredChains*(self: Service, address, preferredChainIds: string): bool =
+    if not self.walletAccountsContainsAddress(address):
+      error "account's address is not among known addresses: ", address=address
+      return false
+    try:
+      var account = self.getAccountByAddress(address)
+      let response = status_go_accounts.updateAccount(account.name, account.address, account.path, account.publicKey,
+        account.keyUid, account.walletType, account.colorId, account.emoji, account.isWallet, account.isChat, preferredChainIds, account.testPreferredChainIds)
+      if not response.error.isNil:
+        error "status-go error", procName="updateWalletAccount", errCode=response.error.code, errDesription=response.error.message
+        return false
+      self.updateAccountInLocalStoreAndNotify(address, name ="", colorId = "", emoji = "", positionUpdated = none(bool), prodPreferredChains = preferredChainIds, testPreferredChains = "")
```

Comment:
This is quite a strange API
Reading this I believe you are setting name to empty 🤔 

It would be better to have a function per attributes set if doing that

Reply:
Hey we only update preferredchains by this and not the name to empty.